presets/workspace/inference/vllm/inference_api.py

```diff
@@ -292,6 +292,24 @@ def try_get_max_available_seq_len(args: argparse.Namespace) -> int | None:
         return None
 
 
+def get_max_gpu_memory_utilization() -> float:
+    # Derive an optimal gpu_memory_utilization from the current CUDA free/total
+    # bytes on the active device. This caps vLLM's GPU usage to what is actually
+    # available at runtime (accounting for driver/context/other processes) to reduce
+    # OOM risk during inference. Floor to 2 decimals to avoid rounding
+    # up. See https://github.com/kaito-project/kaito/issues/1374.
+    free_memory, total_memory = torch.cuda.mem_get_info()
+
+    # Reserve 1.5 GiB for vLLM runtime overhead to prevent OOM errors during initialization.
+    # This buffer is critical for smaller GPUs (e.g. A10) where memory margins are tight.
+    free_memory -= 1.5 * 1024**3
```

Comment:
This buffer was reserved for pytorch memory fragments. 
I didn't notice the memory reserved for cuda driver. So, I think now we don't need to reserve an extra 1.5G vram. maybe 500M is enough

Reply:
Still encountering OOM with 512M 😞